ext/drmgr/drmgr.c

```diff
@@ -1718,7 +1757,12 @@ drmgr_thread_init_event(void *drcontext)
     for (i = 0; i < iter.num; i++) {
         if (!iter.cbs.generic[i].pri.valid)
             continue;
-        (*iter.cbs.generic[i].cb.thread_cb)(drcontext);
+        void *usr_data = iter.cbs.generic[i].usr_data;
+        if (usr_data == NULL)
+            (*iter.cbs.generic[i].cb.thread_cb.cb_no_usr_data)(drcontext);
```

Comment:
Hmm, this seems unsafe: if I call drmgr_register_thread_init_event_usr_data(myfunc, NULL) and then in myfunc I check the value of the user_data param I'll be looking at an uninitialized value.